### PR TITLE
Standardised the recover id object workflow

### PR DESCRIPTION
**Pull request by lassemand**
Small note: As I am not an admin I am unable to make tests of this workflow (As I would not either be able to use the environment or use the secrets), and hence this is untested.

## Purpose

We do not want to have deviating workflows. 

Another point being that I want to make a matrix to do this instead of multiple release flows. This is the first step towards that for easier reviewing

## Changes

Rename release.yml -> release-recover-id-object.yml
Setup the workflows to be alike the rest of the release flows. 

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Comment by DOBEN** [comment]
I don't think we want a matrix in the release flows. We release tools only if needed (e.g. some new features have been implemented). The projects in the `misc_tools` are rather unrelated to each other so no common (matrix-like) release is done.

File: .github/workflows/release-recover-id-object.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: Publish Docker image for recover-id-object to DockerHub
+
+on:
```
Comment (DOBEN) [inline .github/workflows/release-recover-id-object.yml:3]: Suggestion: Include a manual trigger which we needed in the past sometimes when tagging was messed up.
```
  workflow_dispatch: # allows manual trigger
```

File: .github/workflows/release-recover-id-object.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            echo "::error $FULL_IMAGE_TAG already exists"
+            exit 1
+          elif [ ! "${{ github.ref_name }}" = "recover-id-object/${VERSION}" ]; then
+             echo "::error Expected tag ${EXPECTED_TAG} does not match the version ${VERSION}."
+             exit 1
+          else
+             # Store the full image tag into a tag variable for the following step.
+             echo "tag=${FULL_IMAGE_TAG}" > "$GITHUB_OUTPUT"
+          fi
+          echo "tag=$FULL_IMAGE_TAG" >> $GITHUB_ENV
```
Comment (DOBEN) [inline .github/workflows/release-recover-id-object.yml:50]: This line seems not be needed. You are using the output from the line 48.

**Review by DOBEN — APPROVED** [review]

File: .github/workflows/release-recover-id-object.yml (hunk 1 above)
Comment (lassemand) [reply .github/workflows/release-recover-id-object.yml:3]: I would like that, but for I consider that out of scope for this PR.

Scope for the PR is strictly to align it with the other releases. So if I did that I would also have to do that in the other releases. :)

File: recover-id-object/Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "recover-id-object"
-version = "2.0.1"
+version = "0.0.4"
```
Comment (lassemand) [inline recover-id-object/Cargo.toml:3]: For testing strictly

File: .github/workflows/release-recover-id-object.yml (hunk 1 above)
Comment (DOBEN) [reply .github/workflows/release-recover-id-object.yml:3]: The `workflow_dispatch` is included in the set up of most other releases e.g:
https://github.com/Concordium/concordium-dapp-examples/tree/main/.github/workflows

Comment (lassemand) [reply .github/workflows/release-recover-id-object.yml:3]: Adding it anyways

**Review by td202 — APPROVED** [review]

**Pull request merged**